include/syncmer.hpp

```diff
@@ -0,0 +1,582 @@
+    //!\brief Compare to the sentinel of the syncmer_view.
+    friend bool operator==(sentinel const & lhs, basic_iterator const & rhs)
+    {
+        return rhs == lhs;
+    }
+
+    //!\brief Compare to the sentinel of the syncmer_view.
+    friend bool operator!=(sentinel const & lhs, basic_iterator const & rhs)
+    {
+        return !(lhs == rhs);
+    }
+
+    //!\brief Compare to the sentinel of the syncmer_view.
+    friend bool operator!=(basic_iterator const & lhs, sentinel const & rhs)
+    {
+        return !(lhs == rhs);
+    }
+    //!\}
+
+    //!\brief Pre-increment.
+    basic_iterator & operator++() noexcept
+    {
+        next_unique_syncmer();
+        return *this;
+    }
+
+    //!\brief Post-increment.
+    basic_iterator operator++(int) noexcept
+    {
+        basic_iterator tmp{*this};
+        next_unique_syncmer();
+        return tmp;
+    }
+
+    //!\brief Return the syncmer.
+    value_type operator*() const noexcept
+    {
+        return syncmer_value;
+    }
+
+private:
+    //!\brief The syncmer value.
+    value_type syncmer_value{};
+
+    //!\brief The offset relative to the beginning of the window where the syncmer value is found.
+    size_t syncmer_position_offset{};
+
+    //!\brief Iterator to the rightmost value of one kmer.
+    urng1_iterator_t urng1_iterator{};
+
+    //!\brief Iterator to the rightmost value of one kmer in the second range.
+    urng2_iterator_t urng2_iterator{};
+
+    //!brief Iterator to last element in range.
+    urng1_sentinel_t urng1_sentinel{};
+
+    //!\brief The number of values in one window.
+    size_t w_size{};
+
+    //!\brief Stored values per window. It is necessary to store them, because a shift can remove the current syncmer.
+    std::deque<value_type> window_values{};
+
+    //!\brief Increments iterator by 1.
+    void next_unique_syncmer()
+    {
+        while (!next_syncmer()) {}
+    }
+
+    //!\brief Returns new window value.
+    auto window_value() const
+    {
+        return *urng1_iterator;
+    }
+
+
+    //!\brief Advances both windows to the next position.
+    void advance_window()
+    {
+        ++urng1_iterator;
+        ++urng2_iterator;
+    }
+
+    //!\brief Calculates syncmers for the first window.
+    void window_first(const size_t kmer_size, const size_t w_size)
+    {
+        if (w_size == 0u)
+            return;
+
+        for (int i = 1u; i < kmer_size - 1 ; ++i)
+        {
+            window_values.push_back(window_value());
+            ++urng1_iterator;
+        }
+        window_values.push_back(window_value());
+
+        auto smallest_s_it = std::ranges::min_element(window_values, std::less<value_type>{});
+        syncmer_position_offset = std::distance(std::begin(window_values), smallest_s_it);
+
+
+        if constexpr (opensyncmer)
+        {
+            if (syncmer_position_offset == 0)
+            {
+                auto syncmer_it = urng2_iterator;
+                syncmer_value = *syncmer_it;
+            };
```

Comment:
Why is there a `;` here?

Reply:
This is still not resolved.

Reply:
You also have these in other positions of the code, I think that you should delete them.